relay-server/src/endpoints/forward.rs

```diff
@@ -75,6 +75,7 @@ impl ResponseError for ForwardedUpstreamRequestError {
                 HttpError::ActixPayload(e) => e.error_response(),
                 HttpError::ActixJson(e) => e.error_response(),
             },
+            UpstreamRequestError::SendFailed(_) => HttpResponse::GatewayTimeout().finish(),
```

Comment:
This should be a `500` instead of a `502`, since it indicates a failure to build the request, correct?

Reply:
this error actually needs more destructuring

Reply:
Can we instead map all relevant errors to `UpstreamRequestError` kinds? It is rather unfortunate having to destructure errors like this since it is too easy to miss cases, as shown by this PR.